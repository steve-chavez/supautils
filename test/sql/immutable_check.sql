create table products (
    product_no integer,
    date timestamptz check (now() > '2025-03-01')
);
\echo

create table products (
    product_no integer,
    name text check (name <> current_setting('request.name', true))
);
