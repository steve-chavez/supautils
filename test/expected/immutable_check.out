create table products (
    product_no integer,
    date timestamptz check (now() > '2025-03-01')
);
ERROR:  usage of non-immutable function in CHECK constraints are forbidden
\echo

create table products (
    product_no integer,
    name text check (name <> current_setting('request.name', true))
);
ERROR:  usage of non-immutable function in CHECK constraints are forbidden
